### ISSUE TITLE
Problem: Debugging `FindLeaks` result correctness issue

### ISSUE THREAD

**Issue by foreseaz** [nleak/nleak#24]
Currently the output is always "no leaks", need to verify if there is something wrong with the whole flow.

**Comment by foreseaz** [comment]
@foreseaz @Frost-Lee @proy-11

**Comment by foreseaz** [comment]
I ran the test this morning, and there are difference between two heaps.

```
{
  "leaks": [],
  "stackFrames": [],
  "sourceFiles": {},
  "heapStats": [
    {
      "numNodes": 43389,
      "numEdges": 180444,
      "totalSize": 3789514,
      "hiddenSize": 879224,
      "arraySize": 469160,
      "stringSize": 1437616,
      "objectSize": 146384,
      "codeSize": 450312,
      "closureSize": 257752,
      "regexpSize": 2688,
      "heapNumberSize": 33312,
      "nativeSize": 76170,
      "syntheticSize": 1504,
      "consStringSize": 29632,
      "slicedStringSize": 288,
      "symbolSize": 5424,
      "unknownSize": 48
    },
    {
      "numNodes": 48588,
      "numEdges": 196488,
      "totalSize": 4155298,
      "hiddenSize": 922320,
      "arraySize": 650520,
      "stringSize": 1527672,
      "objectSize": 149640,
      "codeSize": 462920,
      "closureSize": 258680,
      "regexpSize": 2688,
      "heapNumberSize": 66304,
      "nativeSize": 76170,
      "syntheticSize": 2352,
      "consStringSize": 30272,
      "slicedStringSize": 288,
      "symbolSize": 5424,
      "unknownSize": 48
    },
    {
      "numNodes": 48134,
      "numEdges": 194715,
      "totalSize": 4107146,
      "hiddenSize": 900280,
      "arraySize": 630056,
      "stringSize": 1522360,
      "objectSize": 149640,
      "codeSize": 462584,
      "closureSize": 258680,
      "regexpSize": 2688,
      "heapNumberSize": 66304,
      "nativeSize": 76170,
      "syntheticSize": 2352,
      "consStringSize": 30272,
      "slicedStringSize": 288,
      "symbolSize": 5424,
      "unknownSize": 48
    }
  ],
  "rankingEvaluation": {
    "leakShare": [],
    "transitiveClosureSize": [],
    "retainedSize": []
  }
}

```

**Comment by proy-11** [comment]
I ran the test with latest changes implemented for CI to run, [PR#31](https://github.com/nleak/nleak/pull/31/files),
nleak generates results successfully showing memory leaks in guest_app, please find the results snapshots below - 

report log - 
<img width="500" alt="Screenshot 2022-10-29 at 10 56 00 AM" src="https://user-images.githubusercontent.com/100389556/198846170-98651f4e-6ed6-444c-ae03-a4efd4693c68.png">

results json - 
<img width="500" alt="Screenshot 2022-10-29 at 10 58 45 AM" src="https://user-images.githubusercontent.com/100389556/198846270-a78adf3b-af52-48ea-817e-25cff19c4c56.png">

Result log file - 
[nleak_report.log](https://github.com/nleak/nleak/files/9894284/bleak_report.log)

On removing the leak from guest_app, and executing nleak again, clean result is getting generated.
